Content.Shared/Throwing/ThrownItemSystem.cs

```diff
@@ -70,7 +70,7 @@ private void HandleCollision(EntityUid uid, ThrownItemComponent component, ref S
 
         private void PreventCollision(EntityUid uid, ThrownItemComponent component, ref PreventCollideEvent args)
         {
-            if (args.OtherEntity == component.Thrower)
+            if (args.OtherEntity == component.Thrower || HasComp<ThrownItemAvoidComponent>(args.OtherEntity))
             {
                 args.Cancelled = true;
             }
```

Comment:
PreventCollideEvent is for 1 entity to not collide with another, not to ignore an entire class of entities. Use collision layers for this.